core/components/atoms/checkbox/Checkbox.tsx

```diff
@@ -134,7 +139,7 @@ export const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>((props
     }
     if (onChange) onChange(e);
   };
-  const id = `${name}-${label}-${uidGenerator()}`;
+  const id = checkboxId ? checkboxId : `${name}-${label}-${uidGenerator()}`;
```

Comment:
We can avoid passing the `checkboxId` as a prop to the checkbox. `checkboxId` is computed in the same way in both the dropdown and checkbox components.